src/hub/vendor/invoices.py

```diff
@@ -78,75 +78,36 @@ def run(self) -> bool:
         Parse the event data sent by Stripe contained within self.payload
         :return True to indicate successfully sent
         """
+        invoice = self.payload.data.object
         logger.debug("invoice payment succeeded", payload=self.payload)
-        subscription_id = self.payload.data.object.subscription
+        subscription_id = invoice.subscription
         subscription = retrieve_stripe_subscription(subscription_id)
         if subscription:
             plan = subscription.get("plan")
-            created = self.payload.data.object.created
             customer = subscription.get("customer")
             user_id = None
             metadata = customer.get("metadata")
             if metadata:
                 user_id = metadata.get("userid")
-            is_subscription_new = self.determine_new(subscription, created)
```

Comment:
The invoice contains the reason for why it was created. If 'billing_reason' is 'subscription_create', then it is a new subscription. Otherwise it is a recurring subscription. The date comparisons that were being done in the `determine_new` method are not necessary to determine if it is a new subscription. This logic is already being used elsewhere in this file to determine if it is a new subscription.